airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLSqlOperations.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.mysql;
+
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.destination.jdbc.DefaultSqlOperations;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MySQLSqlOperations extends DefaultSqlOperations {
+
+  private boolean isLocalFileEnabled = false;
+
+  @Override
+  public void executeTransaction(JdbcDatabase database, List<String> queries) throws Exception {
+    database.executeWithinTransaction(queries);
+  }
+
+  @Override
+  public void insertRecords(JdbcDatabase database,
```

Comment:
Readability suggestion:

1) Push the logic to check for enabled mode into it's own function.
2) Separate creating and clearing the tmp file from `loadDataIntoTable` to avoid the possibility of throwing an exception from a `finally` block. Catch all exceptions in `loadDataIntoTable` and log it instead.

Main thing is to avoid the pattern of throwing an exception from the finally block since it can potentially swallow the exception from the main try block. What do you think?

```
 @Override
  public void insertRecords(JdbcDatabase database,
                            List<AirbyteRecordMessage> records,
                            String schemaName,
                            String tmpTableName)
      throws SQLException {
    if (records.isEmpty()) {
      return;
    }
    verifyLocalFileEnabled(database);
    
    File tmpFile = null;
    try {
      tmpFile = Files.createTempFile(tmpTableName + "-", ".tmp").toFile();
    } catch (IOException e) {
      throw new SQLException(e);
    }

    loadDataIntoTable(database, records, schemaName, tmpTableName, tmpFile);

    try {
      if (tmpFile != null) {
        Files.delete(tmpFile.toPath());
      }
    } catch (IOException e) {
      throw new SQLException(e);
    }

  }

  private void verifyLocalFileEnabled(JdbcDatabase database) throws SQLException {
    boolean localFileEnabled = isLocalFileEnabled || checkIfLocalFileIsEnabled(database);

    if (!localFileEnabled) {
      tryEnableLocalFile(database);
    }
    
    isLocalFileEnabled = true;
  }

  private void loadDataIntoTable(JdbcDatabase database,
                                 List<AirbyteRecordMessage> records,
                                 String schemaName,
                                 String tmpTableName,
                                 File tmpFile)
      throws SQLException {
    database.execute(connection -> {
      try {
        writeBatchToFile(tmpFile, records);

        String absoluteFile = "'" + tmpFile.getAbsolutePath() + "'";

        String query = String.format(
            "LOAD DATA LOCAL INFILE %s INTO TABLE %s.%s FIELDS TERMINATED BY ',' ENCLOSED BY '\"' LINES TERMINATED BY '\\r\\n'",
            absoluteFile, schemaName, tmpTableName);

        try (Statement stmt = connection.createStatement()) {
          stmt.execute(query);
        }
        
      } catch (Exception e) {
          Logger.error(e);
      }
    });
  }```